adaptive_scheduler/models.py

```diff
@@ -402,6 +453,75 @@ def has_windows(self):
     def n_windows(self):
         return self.windows.size()
 
+    def cache_airmasses_within_kernel_windows(self, kernel_intervals_for_resources, network_model, semester_start, interval_size=30*60):
```

Comment:
This is the main function that computes the airmass values, normalizes them, and then caches them along with the unix timestamps they correspond too so they can be interpolated in the kernel. There is a slight complication in that we can now have multiple different targets. We actually ignore this in the observation portal right now and just use the first configurations target for airmass, but here I opted to take an average of all the targets, since that seems at least slightly better than ignoring most targets. Ideally we would factor in each configurations duration to vary the weighting of the different targets airmasses over the time window but its too complicated for me to figure out right now so this seems like a good enough approximation right now.